### ISSUE TITLE
Pro Version S3 Issues -  Continuous replay and non trigger of middleware events

### ISSUE THREAD

**Issue by pran-m** [buger/goreplay#708]
I'm using the GoReplay Pro Version.  I'm facing below issues when replaying through S3. 

1) Every minute a new S3 File is created and stored in the bucket. I want to replay as and when a new file arrives in the bucket.  While replaying, it is not detecting new files arriving at the bucket. It only replays the existing files when the GoReplay command starts. How to approach this issue?

Command to Record:
sudo ./gor --input-raw-track-response --output-file-append --input-raw :8000 --output-file s3://<bucket-name>/requests-%Y-%m-%d-%H-%M.gor

Command to Replay:
sudo ./gor --middleware "node ./middleware.js" --input-file s3://<bucket_name>/requests- 
output-http http://localhost:8001

2)  Second issue is I'm using a goreplay_middleware in Node.Js to do some transformation.  gor.on("replay)"event is not triggered when replaying through S3 however it works file when I replay the local GOR files.  

This is how I use the middleware,
const gor = require("goreplay_middleware");
gor.on("request", function (req) {
    gor.on("response", req.ID, function (resp) {
        gor.on("replay", req.ID, function (repl) { // This block is not triggered.

Command to Replay:
sudo ./gor --middleware "node ./middleware.js" --input-file s3://<bucket_name>/requests- 
output-http http://localhost:8001

**Comment by buger** [comment]
That's true. At the moment if you are using S3, you can't replay it in live mode. 
One of the alternatives here will using `input-tcp` and `output-tcp`, e.g. there will be "collector" agents forwarding data to the instance which does the replay. And if you want to save data to s3 for backup, you can also add multiple outputs. 

See this doc https://github.com/buger/goreplay/wiki/Distributed-configuration

**Comment by pran-m** [comment]
Thanks. Let me check other alternatives you suggested..

There is another issue regarding middleware not triggering "replay" event which I mentioned along with this issue.  Middleware and replay event is critical for our application as we compare replay response and original response here. 

Please , do you have any update on this?
